### PR TITLE
Do not set the default index cleaner, rollover and dependencies image in CR

### PR DESCRIPTION
**Pull request by objectiser**
Signed-off-by: Gary Brown <gary@brownuk.com>

Previously the persisted CR for simplest,yaml included:
```
    Es Index Cleaner:
      Image:           jaegertracing/jaeger-es-index-cleaner:1.17.1
      Number Of Days:  7
      Resources:
      Schedule:  55 23 * * *
    Es Rollover:
      Image:  jaegertracing/jaeger-es-rollover:1.17.1
      Resources:
      Schedule:  0 0 * * *
```
Now
```
    Es Index Cleaner:
      Number Of Days:  7
      Resources:
      Schedule:  55 23 * * *
    Es Rollover:
      Resources:
      Schedule:  0 0 * * *
```

### CODE REVIEW

**Comment by pavolloffay** [comment]
Could you please explain what issues were this causing? Or what is it needed?

**Comment by objectiser** [comment]
@pavolloffay The problem is that these image fields are not being upgraded, e.g.
```
    storage:
      cassandraCreateSchema: {}
      dependencies:
        image: jaegertracing/spark-dependencies
        resources: {}
        schedule: 55 23 * * *
      elasticsearch:
        nodeCount: 1
        redundancyPolicy: ZeroRedundancy
        resources:
          requests:
            memory: 1Gi
        storage: {}
      esIndexCleaner:
        enabled: true
        image: registry.redhat.io/distributed-tracing/jaeger-es-index-cleaner-rhel7:1.13.1
        numberOfDays: 7
        resources: {}
        schedule: 55 23 * * *
      esRollover:
        image: jaegertracing/jaeger-es-rollover:1.13.1
        resources: {}
        schedule: '*/30 * * * *'
      options: {}
      type: elasticsearch
    strategy: production
...
  status:
    phase: ""
    version: 1.17.1
```
I need to also fix the spark dependencies image.

**Comment by pavolloffay** [comment]
Was the CR originally created with Jaeger operator 1.13.1?

File: pkg/strategy/controller.go [hunk 1]
```diff
@@ -144,7 +143,6 @@ func normalizeIndexCleaner(spec *v1.JaegerEsIndexCleanerSpec, storage string) {
 		trueVar := true
 		spec.Enabled = &trueVar
 	}
-	spec.Image = util.ImageName(spec.Image, "jaeger-es-index-cleaner-image")
```
Comment (pavolloffay) [inline pkg/strategy/controller.go:147]: I am wondering if there should be a null check here.

Comment (objectiser) [reply pkg/strategy/controller.go:147]: It will never return a null, as if not specified it is initialised from the cli flag, which has a default value.

**Comment by objectiser** [comment]
@pavolloffay yes was originally created with 1.13.1 and then upgraded via OLM.

**Comment by objectiser** [comment]
@kevinearls Would it be possible to do a test on this PR? GH CI seems to be causing issues with cassandra test - timing out.

File: pkg/strategy/controller.go (hunk 1 above)
Comment (pavolloffay) [reply pkg/strategy/controller.go:147]: I mean we should have been checking `spec.Image` whether is empty before assigning a value to it.

**Comment by objectiser** [comment]
Still failing with internal CI, so needs investigation.

**Comment by objectiser** [comment]
@pavolloffay Would you mind doing a quick re-review - codecov doesn't seem related to this PR ?

File: pkg/cronjob/spark_dependencies.go [hunk 2]
```diff
@@ -50,6 +52,12 @@ func CreateSparkDependencies(jaeger *v1.Jaeger) *batchv1beta1.CronJob {
 
 	commonSpec := util.Merge([]v1.JaegerCommonSpec{jaeger.Spec.Storage.Dependencies.JaegerCommonSpec, jaeger.Spec.JaegerCommonSpec, baseCommonSpec})
 
+	image := jaeger.Spec.Storage.Dependencies.Image
+	if image == "" {
+		// the version is not included, there is only one version - latest
+		image = viper.GetString("jaeger-spark-dependencies-image")
```
Comment (pavolloffay) [inline pkg/cronjob/spark_dependencies.go:58]: > codecov doesn't seem related to this PR ?

This does not seem to be covered.

File: pkg/cronjob/spark_dependencies.go [hunk 3]
```diff
@@ -50,6 +52,12 @@ func CreateSparkDependencies(jaeger *v1.Jaeger) *batchv1beta1.CronJob {
 
 	commonSpec := util.Merge([]v1.JaegerCommonSpec{jaeger.Spec.Storage.Dependencies.JaegerCommonSpec, jaeger.Spec.JaegerCommonSpec, baseCommonSpec})
 
+	image := jaeger.Spec.Storage.Dependencies.Image
+	if image == "" {
```
Comment (pavolloffay) [inline pkg/cronjob/spark_dependencies.go:56]: Why isn't this file using `util.ImageName`? The  `es_index_cleaner` is using is in this PR.

File: pkg/cronjob/spark_dependencies.go (hunk 2 above)
Comment (objectiser) [reply pkg/cronjob/spark_dependencies.go:58]: This is tested in [both of these tests](https://github.com/jaegertracing/jaeger-operator/blob/e117f178679213b5bd595f37df43c86961dffbe7/pkg/cronjob/spark_dependencies_test.go#L207-L226).

File: pkg/cronjob/spark_dependencies.go (hunk 3 above)
Comment (objectiser) [reply pkg/cronjob/spark_dependencies.go:56]: The `util.ImageName` resolves the image name but also appends the current Jaeger version. As the dependencies image is the only one that does not have a tag, the image is just resolved here.

Comment (pavolloffay) [reply pkg/cronjob/spark_dependencies.go:56]: :+1: could you please add a comment to the source code?

Comment (objectiser) [reply pkg/cronjob/spark_dependencies.go:59]: done
